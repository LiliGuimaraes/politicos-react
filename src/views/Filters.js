/*
 * Copyright (c) 2016, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { Drawer, Button } from 'material-ui'

import {
  changePoliticiansList,
  resetPoliticiansList,
} from '../select/politiciansDuck'
import {
  SelectCity,
  SelectEducation,
  SelectElected,
  SelectElection,
  SelectGender,
  SelectMaritalStatus,
  SelectOccupation,
  SelectPoliticalOffice,
  SelectPoliticalParty,
  SelectPolitician,
  SelectState,
} from '../select'

export class Filters extends Component {
  onChangeQuery() {
    const state = this.props.state
    let queryString = new URLSearchParams()

    // FIXME: only select component
    Object.keys(state).map(objectKey => {
      const obj = state[objectKey]
      if (!obj.selected) {
        return
      }
      obj.selected.map(item => {
        return queryString.set(obj.query, item.value)
      })
    })

    if (queryString.toString() == '') {
      return
    }

    this.props.dispatch(resetPoliticiansList())

    this.props.HTTPClient.get('/politicians?' + queryString.toString()).then(
      result => {
        this.props.dispatch(changePoliticiansList(result.data))
      }
    )
  }

  render() {
    return (
      <Drawer open type="permanent" anchor="right">
        <div style={{ width: 340, margin: '64px 24px', paddingBottom: 72 }}>
          <SelectPolitician {...this.props} />
          <SelectPoliticalParty {...this.props} />
          <SelectPoliticalOffice {...this.props} />
          <SelectEducation {...this.props} />
          <SelectElection {...this.props} />
          <SelectState {...this.props} />
          <SelectCity {...this.props} />
          <SelectElected {...this.props} />
          <SelectGender {...this.props} />
          <SelectOccupation {...this.props} />
          <SelectMaritalStatus {...this.props} />
        </div>
        <div
          style={{
            position: 'fixed',
            bottom: 0,
            background: '#eee',
            padding: 24,
            width: '100%',
          }}
        >
          <Button
            raised
            color="primary"
            onClick={this.onChangeQuery.bind(this)}
          >
            Filtrar
          </Button>
        </div>
      </Drawer>
    )
  }
}

/* istanbul ignore next */
const mapStateToProps = state => {
  return {
    state,
  }
}

Filters.propTypes = {
  state: PropTypes.object,
  dispatch: PropTypes.func,
  HTTPClient: PropTypes.object,
}

export default connect(mapStateToProps)(Filters)
