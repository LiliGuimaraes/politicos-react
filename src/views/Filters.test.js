/*
 * Copyright (c) 2017, Marcelo Jorge Vieira <metal@alucinados.com>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import React from 'react'
import renderer from 'react-test-renderer'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'

import { Filters } from './Filters'
import {
  changePoliticiansList,
  resetPoliticiansList,
} from '../select/politiciansDuck'

describe('<Filters />', () => {
  const mockStore = configureMockStore()
  const storeStateMock = {
    city: { list: { objects: [] } },
    education: { list: { objects: [] } },
    elected: { list: [] },
    election: { list: { objects: [] } },
    gender: { list: [] },
    maritalStatus: { list: { objects: [] } },
    occupation: { list: { objects: [] } },
    politicalOffice: { list: { objects: [] } },
    politicalParty: { list: { objects: [] } },
    politician: { list: { objects: [] } },
    state: { list: { objects: [] } },
  }
  const store = mockStore(storeStateMock)

  it('renders without crashing', () => {
    const HTTPClient = { get: () => Promise.resolve({}) }
    const component = renderer.create(
      <Provider store={store}>
        <Filters HTTPClient={HTTPClient} />
      </Provider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should not call dispatches when there is no selected item', () => {
    const state = { metal: {} }
    const dispatchMock = jest.fn()
    const HTTPClient = { get: () => Promise.resolve({}) }
    const component = mount(
      <Provider store={store}>
        <Filters
          dispatch={dispatchMock}
          state={state}
          HTTPClient={HTTPClient}
        />
      </Provider>
    )
    component.find('button').simulate('click')

    expect(dispatchMock).toHaveBeenCalledTimes(0)
  })

  it('should call dispatches when there is selected item', () => {
    const state = { metal: { selected: [{ value: 123 }], query: 'q' } }
    const result = { data: { metal: 123 } }
    const HTTPClient = {
      get: jest.fn(() => {
        return {
          then: fn => fn(result),
        }
      }),
    }
    const dispatchMock = jest.fn()
    const component = mount(
      <Provider store={store}>
        <Filters
          dispatch={dispatchMock}
          state={state}
          HTTPClient={HTTPClient}
        />
      </Provider>
    )
    component.find('button').simulate('click')

    expect(dispatchMock).toHaveBeenCalledWith(resetPoliticiansList())
    expect(dispatchMock).toHaveBeenCalledWith(
      changePoliticiansList(result.data)
    )
  })
})
