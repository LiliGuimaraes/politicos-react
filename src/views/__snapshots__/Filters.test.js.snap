// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filters /> renders without crashing 1`] = `
<div
  className="MuiDrawer-docked-1"
>
  <div
    className="MuiPaper-root-12 MuiPaper-shadow0-14 MuiDrawer-paper-2 MuiDrawer-paperAnchorRight-4 MuiDrawer-paperAnchorDockedRight-9"
  >
    <div
      style={
        Object {
          "margin": "64px 24px",
          "paddingBottom": 72,
          "width": 340,
        }
      }
    >
      <div>
        <div>
          <h4>
            Pol√≠ticos
          </h4>
          <div
            className="Select Select--multi is-clearable is-loading is-searchable"
            style={undefined}
          >
            <div
              className="Select-control"
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={undefined}
            >
              <span
                className="Select-multi-value-wrapper"
                id="react-select-2--value"
              >
                <div
                  className="Select-placeholder"
                >
                  Loading...
                </div>
                <div
                  className="Select-input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-activedescendant="react-select-2--value"
                    aria-describedby={undefined}
                    aria-expanded="false"
                    aria-haspopup="false"
                    aria-label={undefined}
                    aria-labelledby={undefined}
                    aria-owns=""
                    className={undefined}
                    id={undefined}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    role="combobox"
                    style={
                      Object {
                        "boxSizing": "content-box",
                        "width": "5px",
                      }
                    }
                    tabIndex={undefined}
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </span>
              <span
                aria-hidden="true"
                className="Select-loading-zone"
              >
                <span
                  className="Select-loading"
                />
              </span>
              <span
                className="Select-arrow-zone"
                onMouseDown={[Function]}
              >
                <span
                  className="Select-arrow"
                  onMouseDown={[Function]}
                />
              </span>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Partidos
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-3--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-3--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Cargos
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-4--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-4--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Escolaridades
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-5--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-5--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Elei√ß√µes
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-6--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-6--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Estados
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-7--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-7--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Cidades
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-8--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-8--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Eleito
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-9--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-9--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Sexo
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-10--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-10--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Profiss√µes
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-11--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-11--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Estado Civil
        </h4>
        <div
          className="Select Select--multi is-clearable is-loading is-searchable"
          style={undefined}
        >
          <div
            className="Select-control"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={undefined}
          >
            <span
              className="Select-multi-value-wrapper"
              id="react-select-12--value"
            >
              <div
                className="Select-placeholder"
              >
                Loading...
              </div>
              <div
                className="Select-input"
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <input
                  aria-activedescendant="react-select-12--value"
                  aria-describedby={undefined}
                  aria-expanded="false"
                  aria-haspopup="false"
                  aria-label={undefined}
                  aria-labelledby={undefined}
                  aria-owns=""
                  className={undefined}
                  id={undefined}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  role="combobox"
                  style={
                    Object {
                      "boxSizing": "content-box",
                      "width": "5px",
                    }
                  }
                  tabIndex={undefined}
                  value=""
                />
                <div
                  style={
                    Object {
                      "height": 0,
                      "left": 0,
                      "overflow": "scroll",
                      "position": "absolute",
                      "top": 0,
                      "visibility": "hidden",
                      "whiteSpace": "pre",
                    }
                  }
                >
                  
                </div>
              </div>
            </span>
            <span
              aria-hidden="true"
              className="Select-loading-zone"
            >
              <span
                className="Select-loading"
              />
            </span>
            <span
              className="Select-arrow-zone"
              onMouseDown={[Function]}
            >
              <span
                className="Select-arrow"
                onMouseDown={[Function]}
              />
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "background": "#eee",
          "bottom": 0,
          "padding": 24,
          "position": "fixed",
          "width": "100%",
        }
      }
    >
      <button
        className="MuiButtonBase-root-54 MuiButton-root-39 MuiButton-raised-46 MuiButton-raisedPrimary-48"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label-41"
        >
          Filtrar
        </span>
        <span
          className="MuiTouchRipple-root-56"
        />
      </button>
    </div>
  </div>
</div>
`;
